modules/core/src/main/java/org/apache/ignite/internal/processors/security/IgniteSecurityProcessor.java

```diff
@@ -125,17 +130,24 @@ public IgniteSecurityProcessor(GridKernalContext ctx, GridSecurityProcessor secP
 
     /** {@inheritDoc} */
     @Override public OperationSecurityContext withContext(UUID subjId) {
-        ClusterNode node = Optional.ofNullable(ctx.discovery().node(subjId))
-            .orElseGet(() -> ctx.discovery().historicalNode(subjId));
+        try {
+            ClusterNode node = Optional.ofNullable(ctx.discovery().node(subjId))
+                .orElseGet(() -> ctx.discovery().historicalNode(subjId));
+
+            SecurityContext res = node != null ? secCtxs.computeIfAbsent(subjId,
+                uuid -> nodeSecurityContext(marsh, U.resolveClassLoader(ctx.config()), node))
+                : secPrc.securityContext(subjId);
 
-        SecurityContext res = node != null ? secCtxs.computeIfAbsent(subjId,
-            uuid -> nodeSecurityContext(marsh, U.resolveClassLoader(ctx.config()), node))
-            : secPrc.securityContext(subjId);
+            if (res == null)
+                throw new IllegalStateException("Failed to find security context for subject with given ID : " + subjId);
```

Comment:
What's the point throw an Exception and catch it two lines later?

Reply:
@nizhikov 
After logging inside the catch block, this exception will be thrown outside where it can be handled.

Reply:
We can log it before throw.

Reply:
@nizhikov fixed